articleController.js

```diff
@@ -1,5 +1,19 @@
 var articleController = {};
 
 articleController.index = function () {
-  blog.determineData(); 
+  blog.determineData();
+};
+
+articleController.category = function (ctx, next) {
+  var categoryData = function (data) { //rename data to articles?  it is the return of .findByCategory
+    ctx.articles = data;
```

Comment:
I'm leaving my last comment because it still stands as an example of how I was confused about what this function is actually trying to do. I think I've found your bug though. Updating soon.